packages/schema/src/jsonSchema.ts

```diff
@@ -0,0 +1,45 @@
+import type { Schema, SearchableType } from '@orama/orama';
+import type { JSONSchema4 } from 'json-schema';
+
+const isJsonObject = (jsonSchema: JSONSchema4) => jsonSchema.type === 'object'
+
+const assertTypeObject = (jsonSchema: JSONSchema4) => {
+    if (!isJsonObject(jsonSchema)) {
+        throw new Error('Provided JSON schema must be an object type');
+    }
+}
+
+const ORAMA_SUPPORTED_TYPES: Set<JSONSchema4['type']> = new Set(['string', 'number', 'boolean'])
+
+const isArraySupportedByOrama = (jsonSchema: JSONSchema4): boolean => {
+    if (jsonSchema.type === 'array' && jsonSchema.items && !Array.isArray(jsonSchema.items)) {
+        return ORAMA_SUPPORTED_TYPES.has(jsonSchema.items.type);
+    }
+    return false
+}
+
+const isSupportedByOrama = (jsonSchema: JSONSchema4): boolean => {
+    return ORAMA_SUPPORTED_TYPES.has(jsonSchema.type) || isArraySupportedByOrama(jsonSchema);
+}
+
+const extractOramaType = (jsonSchema: JSONSchema4): SearchableType => {
+    const oramaType = ORAMA_SUPPORTED_TYPES.has(jsonSchema.type) ? jsonSchema.type : `${(jsonSchema.items as JSONSchema4)!.type}[]`
+
+    return oramaType as SearchableType
+}
+
+export const schemaFromJson = (jsonSchema: JSONSchema4) => {
+    assertTypeObject(jsonSchema)
+
+    const oramaSchema: Schema = {}
+
+    for (const [propertyName, propertyDefinition] of Object.entries(jsonSchema.properties || {})) {
+        if (isSupportedByOrama(propertyDefinition)) {
+            oramaSchema[propertyName] = extractOramaType(propertyDefinition)
+        } else if (isJsonObject(propertyDefinition)) {
+            oramaSchema[propertyName] = schemaFromJson(propertyDefinition)
+        }
```

Comment:
should we throw an error here?

Reply:
Does orama throws if it receive a wrong schema? If not, I would not throw here either

Reply:
Looking at the `validateSchema` function provided by Orama, it's ignoring the invalid types, I would do the same here